node/pkg/watchers/evm/connectors/batch_poller.go

```diff
@@ -155,15 +155,28 @@ func (b *BatchPollConnector) pollBlocks(ctx context.Context, logger *zap.Logger,
 				if err != nil {
 					return prevBlocks, fmt.Errorf("failed to get gap blocks: %w", err)
 				}
+				prevBlock := prevBlocks[idx]
+				errorFound := false
 				for _, block := range gapBlocks {
+					if block.Number.Uint64() == 0 {
+						errorFound = true
+						break
+					}
+
 					b.blockFeed.Send(block)
+					prevBlock = block
+				}
+
+				if errorFound {
+					newBlocks[idx] = prevBlock
```

Comment:
can you add a comment with what this code effectively does on errors?

Reply:
if I'm reading it correctly, it appears to not move that block (tag) forward if it hits a block which had an error. but should it just break instead of setting it back to the previous block (before any of these gap blocks) and continuing. maybe it's worth detailing an example

Reply:
It sets it to the last successfully read block so that the next iteration will pick up after that. We have already published blocks up to that point, so I don't think we want to revert back before blocks we already published.

Reply:
Made the revisions we discussed offline.